### PR TITLE
[14.0][IMP] payroll: Only payroll manager can edit salary rules

### PR DESCRIPTION
**Pull request by norlinhenrik**
See discussion in #88 
What do you think about this solution?

### CODE REVIEW

**Comment by nimarosa** [comment]
Looks good solution for me. Maybe I would suggest to call the new group "Payroll Superadmin" or "Payroll Admin" since this group will have all permissions too and can be used to limit and improve the other roles in the future. 

But looks good to me. 👍

**Comment by nimarosa** [comment]
@norlinhenrik Also check pre commit failing .

**Comment by mtelahun** [comment]
I don't understand what the point of this new group is. Having an Officer and a Manager makes sense. You can divide responsibilities for running payroll and configuring payroll. A new group that has a Manager's permissions and has only the additional permission of writing python code just doesn't make any sense to me. It's simpler and easier just to keep it a responsibility of the payroll manager.

File: payroll/models/hr_payslip.py [hunk 1]
```diff
@@ -623,9 +623,13 @@ def get_lines_dict(self):
                 for rule in payslip._get_salary_rules():
                     localdict = rule._reset_localdict_values(localdict)
                     # check if the rule can be applied
-                    if rule._satisfy_condition(localdict) and rule.id not in blacklist:
+                    rule_sudo = rule.sudo()
```
Comment (mtelahun) [inline payroll/models/hr_payslip.py]: Why is this running with admin privileges? If anything, doesn't this make it less secure? Before it was running as Payroll Manager now it's running as Administrator.

Comment (nimarosa) [reply payroll/models/hr_payslip.py]: I didn't saw this change. Why is this sudo necessary now?

**Comment by norlinhenrik** [comment]
Ok I made it very simple now.

**Review by nimarosa — APPROVED** [review]

**Pull request merged**